components/public-api-server/pkg/apiv1/tokens_test.go

```diff
@@ -65,18 +65,39 @@ func TestTokensService_CreatePersonalAccessTokenWithoutFeatureFlag(t *testing.T)
 
 func TestTokensService_GetPersonalAccessToken(t *testing.T) {
 	user := newUser(&protocol.User{})
+	user2 := newUser(&protocol.User{})
 
 	t.Run("permission denied when feature flag is disabled", func(t *testing.T) {
-		serverMock, _, client := setupTokensService(t, withTokenFeatureDisabled)
+		serverMock, dbConn, client := setupTokensService(t, withTokenFeatureDisabled)
+
+		tokens := dbtest.CreatePersonalAccessTokenRecords(t, dbConn,
+			dbtest.NewPersonalAccessToken(t, db.PersonalAccessToken{
+				UserID: uuid.MustParse(user.ID),
+			}),
+			dbtest.NewPersonalAccessToken(t, db.PersonalAccessToken{
+				UserID: uuid.MustParse(user2.ID),
+			}),
+		)
 
 		serverMock.EXPECT().GetLoggedInUser(gomock.Any()).Return(user, nil)
 
-		_, err := client.GetPersonalAccessToken(context.Background(), connect.NewRequest(&v1.GetPersonalAccessTokenRequest{
-			Id: uuid.New().String(),
+		response, err := client.GetPersonalAccessToken(context.Background(), connect.NewRequest(&v1.GetPersonalAccessTokenRequest{
+			Id: tokens[0].ID.String(),
 		}))
 
-		require.Error(t, err, "This feature is currently in beta. If you would like to be part of the beta, please contact us.")
-		require.Equal(t, connect.CodePermissionDenied, connect.CodeOf(err))
+		require.NoError(t, err)
+
+		requireEqualProto(t, &v1.GetPersonalAccessTokenResponse{
+			Token: &v1.PersonalAccessToken{
+				Id:             response.Msg.Token.Id,
+				Hash:           response.Msg.Token.Hash,
+				Name:           response.Msg.Token.Name,
+				Description:    response.Msg.Token.Description,
+				ExpirationTime: response.Msg.Token.ExpirationTime,
+				Scopes:         response.Msg.Token.Scopes,
+				CreatedAt:      response.Msg.Token.CreatedAt,
+			},
+		}, response.Msg)
```

Comment:
After the rebase, you can now reuse the `personalAccessTokenToAPI` method. See https://github.com/gitpod-io/gitpod/pull/14832/files#diff-faefbff9f99df138bade476286958b2870dacd62989297aeef83b576e5954b8bR182